### PR TITLE
Add presentations, essays, and history

### PR DESCRIPTION
**Pull request by brainwane**

### CODE REVIEW

File: source/presentations.rst [hunk 1]
```diff
@@ -25,7 +25,38 @@ Presentations & Articles
...
+* PyGotham, October 2017
+
+  * `Running Vintage Software: PyPI's Aging Codebase
+    <http://pyvideo.org/pygotham-2017/running-vintage-software-pypis-aging-codebase.html>`_
+
+* North Bay Python, December 2017
+
+  * `Running Vintage Software: PyPI's Aging Codebase
+    <https://2017.northbaypython.org/schedule/presentation/5/>`_
+
```
Comment (di) [inline source/presentations.rst]: I think Mahmoud Hashemi's talk "The Packaging Gradient" might be worthwhile to add here: https://www.youtube.com/watch?v=iLVNWfPWAC8

Also the corresponding essay: http://sedimental.org/the_packaging_gradient.html

Comment (brainwane) [reply source/presentations.rst]: Thanks! Adding that and adding further history -- will change PR title back to remove "WIP:" prefix and ask for re-review when it's ready.

**Comment by brainwane** [comment]
Ready for review, @jonparrott @ncoghlan @dstufft

**Comment by ncoghlan** [comment]
Thanks @brainwane!

**Pull request merged**